src/commands/NFD.ts

```diff
@@ -0,0 +1,1003 @@
+  @SlashGroup('nfd')
+  async rename(
+    @SlashOption('name', {
+      type: ApplicationCommandOptionType.String,
+      required: true,
+      description: 'The *existing* name for the NFD.',
+    })
+    @SlashOption('replacement', {
+      type: ApplicationCommandOptionType.String,
+      required: true,
+      description: 'The *new* name for the NFD.',
+    })
+    name: string,
+    replacement: string,
+    interaction: CommandInteraction
+  ) {
+    // Sanity check the new name. Only alphanumeric characters allowed
+    if (
+      replacement.length < this.MIN_NFD_NAME_LENGTH ||
+      replacement.length > this.MAX_NFD_NAME_LENGTH ||
+      replacement.match(/[^a-zA-Z0-9]/g)
+    ) {
+      return interaction.reply({
+        content: `That name is bad. Names must be ${this.MIN_NFD_NAME_LENGTH}-${this.MAX_NFD_NAME_LENGTH} alphanumeric characters.`,
+        ephemeral: true,
+      })
+    }
+
+    // Check the user's cooldowns.
+    const user = await this.client.nFDEnjoyer.findUnique({ where: { id: interaction.user.id } })
+    if (!user) {
+      return interaction.reply({
+        content: "It seems you don't exist in the database. Try minting something first!",
+        ephemeral: true,
+      })
+    }
+    if (user.lastRename.getTime() + this.RENAME_COOLDOWN > Date.now()) {
+      return interaction.reply({
+        content: `Please wait. You can rename again <t:${Math.round(
+          (user.lastRename.getTime() + this.RENAME_COOLDOWN) / 1000
+        )}:R>.`,
+        ephemeral: true,
+      })
+    }
+
+    // Confirm the NFD exists
+    const nfd = await this.getNFDByName(name)
+    if (!nfd) {
+      return interaction.reply({ content: "I couldn't find an NFD with that name.", ephemeral: true })
+    }
+
+    // Confirm that the caller owns the NFD
+    if (nfd.owner != interaction.user.id) {
+      return interaction.reply({ content: "You can't rename something you don't own!", ephemeral: true })
+    }
+
+    // Confirm that no NFD already exists with that name
+    const existing = await this.client.nFDItem.findUnique({
+      where: {
+        name: replacement,
+      },
+    })
+    if (existing) {
+      return interaction.reply({ content: 'An NFD already exists with that name.', ephemeral: true })
+    }
+
+    // All checks passed, update the record and announce it.
+    await this.client.nFDItem.update({
+      where: {
+        name: name,
+      },
+      data: {
+        name: replacement,
+      },
+    })
+
+    // If this was the user's favourite NFD, update the record to the new name
+    const favourite = user.favourite == nfd.name ? replacement : user.favourite
+    await this.client.nFDEnjoyer.update({
+      where: {
+        id: user.id,
+      },
+      data: {
+        lastRename: new Date(),
+        favourite: favourite,
+      },
+    })
+
+    return interaction.reply({ content: `${interaction.user} renamed **${name}** to **${replacement}**!` })
```

Comment:
Another place to prevent pings on the reply

Reply:
Have done, but I figured seeing as this will only ever ping the caller, it wasn't a huge sin. They 'opted in' to an immediate ping back.